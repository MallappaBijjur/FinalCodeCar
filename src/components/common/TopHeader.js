import React from 'react';
import logo from '../../auto1logo.jpg';
import './header.css';

const TopHeader = (props) => (
  <header className='header'>
    <div>
      <img
        className='header__logo'
        src={logo} alt='Auto1LOgo'
      />
    </div>
    <div className='header__nav' data-testid="header">
      <a href='#purchase' className='header__navlink'>Purchase</a>
      <a href='#myOrders' className='header__navlink'>My Orders</a>
      <a href='#Sell' className='header__navlink'>Sell</a>
    </div>
  </header>
)

export default TopHeader;
