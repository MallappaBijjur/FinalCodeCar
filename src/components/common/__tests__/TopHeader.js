import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import rerender from 'react-test-renderer';
import TopHeader from '../TopHeader';

describe("TopHeader", () => {

	const originalError = console.error
	beforeAll(() => {
		console.error = (...args) => {
			if (/Warning.*not wrapped in act/.test(args[0])) {
				return
			}
			originalError.call(console, ...args)
		}
	})

	afterAll(() => {
		console.error = originalError
	});

	afterEach(cleanup);
	
	it("renders without crashing", ()=>{
		const div = document.createElement("div");
		ReactDOM.render(<TopHeader />, div);
	});
	
	it('renders Header component', () => {
        const { getByText, getByTestId } = render(<TopHeader />);   
        const linkElement = getByText(/Purchase/i);
        const elem = getByTestId("header"); 
        expect(linkElement).toBeInTheDocument();
        expect(elem.children.length).toBe(3);
	});
	
	it("matches snapshot", ()=> {
		const tree = rerender.create(<TopHeader />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});




