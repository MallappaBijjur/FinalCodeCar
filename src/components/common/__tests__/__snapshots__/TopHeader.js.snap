// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopHeader matches snapshot 1`] = `
<header
  className="header"
>
  <div>
    <img
      alt="Auto1LOgo"
      className="header__logo"
      src="auto1logo.jpg"
    />
  </div>
  <div
    className="header__nav"
    data-testid="header"
  >
    <a
      className="header__navlink"
      href="#purchase"
    >
      Purchase
    </a>
    <a
      className="header__navlink"
      href="#myOrders"
    >
      My Orders
    </a>
    <a
      className="header__navlink"
      href="#Sell"
    >
      Sell
    </a>
  </div>
</header>
`;
